content/python/concepts/built-in-functions/terms/iter/iter.md

```diff
@@ -0,0 +1,73 @@
+---
+Title: 'iter()'
+Description: 'create an iterator object.'
+Subjects:
+  - 'computer science'
+  - 'data science'
+Tags:
+  - 'Iterators'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+**iter()** function is a built-in function in python that returns an [iterator](https://www.codecademy.com/resources/docs/python/iterators)
+
+## Syntax
+
+```pseudo
+iter(object)
+```
+
+```pseudo
+iter(object, sentinel)
+```
+
+- `object` parameter supplies the object on which the iter() method will return the iterator. The type of this parameter depends on the presence of the optional second `sentinel` parameter. In the case where the sentinel parameter is not given, the object parameter type must be one that supports iteration or a sequence. These types include: list, str, and tuple among others. You may also pass in objects that you've defined with either **\_\_iter\_\_()** or **\_\_getitem\_\_()** methods. When the sentinel parameter is present, the object parameter must be callable.
+
+- `sentinel` is the parameter that stores the value which represents the end of the sequence. When the callable object returns this value, the iterator will terminate.
+
+## Example
+
+```py
+from functools import partial
+import random
+
+## example one iter() with no sentinel parameter
+fruits = ['apples', 'bananas', 'oranges']
+
+my_iterator = iter(fruits)
+print(my_iterator)
+
+## example two iter() with a sentinel parameter
+func_iterator = iter(partial(random.randint, 1, 10), 7)
+print(func_iterator)
```

Comment:
```suggestion
from functools import partial
import random

# iter() with no sentinel parameter
fruits = ['apples', 'bananas', 'oranges']
my_iterator = iter(fruits)
print(next(my_iterator))
print(next(my_iterator))
print(next(my_iterator))
```

Reply:
The reason for this is we want to show a simple example on how to use iter and current example show the location which can be confusing